org.adempiere.base/src/org/compiere/process/SvrProcess.java

```diff
@@ -727,20 +728,17 @@ private void unlock ()
 			//clear interrupt signal so that we can unlock the ad_pinstance record
 			if (Thread.currentThread().isInterrupted())
 				Thread.interrupted();
+			
+			if(m_pi.getAD_PInstance_ID() > 0) {
+				MPInstance mpi = new MPInstance (getCtx(), m_pi.getAD_PInstance_ID(), null);

```

Comment:
Should keep the original if(mpi.get_ID() ==0) check here.

Reply:
Can `mpi.get_ID() == 0` if `m_pi.getAD_PInstance_ID() > 0` ?

Reply:
Ok, reverted